pilot/pkg/serviceregistry/kube/controller/ambient/waypoints.go

```diff
@@ -245,8 +253,95 @@ func (a *index) WaypointsCollection(
 			trafficType = tt
 		}
 
-		return a.makeWaypoint(ctx, gateway, gatewayClass, serviceAccounts, trafficType)
-	}, opts.WithName("Waypoints")...)
+		return makeWaypoint(gateway, gatewayClass, serviceAccounts, trafficType, networkGetter(ctx))
+	}
+}
+
+func GlobalWaypointsCollection(
+	localCluster *multicluster.Cluster,
+	localWaypoints krt.Collection[Waypoint],
+	clusters krt.Collection[*multicluster.Cluster],
+	gatewayClasses krt.Collection[*v1beta1.GatewayClass],
+	globalNetworks networkCollections,
+	opts krt.OptionsBuilder,
+) krt.Collection[krt.Collection[Waypoint]] {
+	return nestedCollectionFromLocalAndRemote(localWaypoints, clusters, func(ctx krt.HandlerContext, c *multicluster.Cluster) *krt.Collection[Waypoint] {
+		pods := c.Pods()
+		krt.Subscribe(ctx, pods)
+		podsByNamespace := krt.NewNamespaceIndex(pods)
+		gateways := c.Gateways()
+		krt.Subscribe(ctx, gateways)
```

Comment:
IIUC this says "if a single Gateway changes, we recreate the clusterWaypoints collection"?

Reply:
I think this is equivalent to what was there before since we're subscribing to that single cluster's gateways. If there's a new gateway added, it might be a waypoint so we do want to subscribe to changes right?

Reply:
Actually nope you're right; I was mixing up subscribing to nested collections vs. subscribing to a flat one. Nice catch!